src/test/java/org/broadinstitute/hellbender/utils/fasta/CachingIndexedFastaSequenceFileUnitTest.java

```diff
@@ -54,120 +58,113 @@ private static long getCacheSize(final long cacheSizeRequested) {
+
+            int middleStart = (contig.getSequenceLength() - querySize) / 2;
+            int middleStop = middleStart + querySize;
+
+            logger.debug(String.format(
+                    "Checking contig %s length %d with cache size %d and query size %d with intermediate query",
+                    contig.getSequenceName(), contig.getSequenceLength(), cacheSize, querySize));
+
+            for (int i = 0; i < contig.getSequenceLength(); i += 10) {
+                int start = i;
+                int stop = start + querySize;
+                if (stop <= contig.getSequenceLength()) {
+                    ReferenceSequence grabMiddle = caching.getSubsequenceAt(contig.getSequenceName(), middleStart,
+                                                                            middleStop);
+                    ReferenceSequence cachedVal = caching.getSubsequenceAt(contig.getSequenceName(), start, stop);
+                    ReferenceSequence uncachedVal = uncached.getSubsequenceAt(contig.getSequenceName(), start, stop);
+
+                    Assert.assertEquals(cachedVal.getName(), uncachedVal.getName());
+                    Assert.assertEquals(cachedVal.getContigIndex(), uncachedVal.getContigIndex());
+                    Assert.assertEquals(cachedVal.getBases(), uncachedVal.getBases());
                 }
             }
         }
-
-        return params.toArray(new Object[][]{});
     }
 
     // make sure some bases are lower case and some are upper case
     @Test
-    public void testMixedCasesInExample() throws FileNotFoundException, InterruptedException {
-        final IndexedFastaSequenceFile original = new IndexedFastaSequenceFile(new File(exampleFASTA));
-        final CachingIndexedFastaSequenceFile casePreserving = new CachingIndexedFastaSequenceFile(
-                IOUtils.getPath(exampleFASTA), true);
-        final CachingIndexedFastaSequenceFile allUpper = new CachingIndexedFastaSequenceFile(IOUtils.getPath(exampleFASTA));
-
-        int nMixedCase = 0;
-        for ( SAMSequenceRecord contig : original.getSequenceDictionary().getSequences() ) {
-            nMixedCase += testCases(original, casePreserving, allUpper, contig.getSequenceName(), -1, -1);
-
-            final int step = 100;
-            for ( int lastPos = step; lastPos < contig.getSequenceLength(); lastPos += step ) {
-                testCases(original, casePreserving, allUpper, contig.getSequenceName(), lastPos - step, lastPos);
+    public void testMixedCasesInExample() throws IOException {
+        try(final IndexedFastaSequenceFile original = new IndexedFastaSequenceFile(new File(exampleFASTA));
+            final CachingIndexedFastaSequenceFile casePreserving = new CachingIndexedFastaSequenceFile(IOUtils.getPath(exampleFASTA), true);
+            final CachingIndexedFastaSequenceFile allUpper = new CachingIndexedFastaSequenceFile(IOUtils.getPath(exampleFASTA));
```

Comment:
Pass in an explicit `false` for `preserveCase` in the constructor call for `allUpper` instead of relying on the defaults.

Reply:
good call, done